src/compute/src/logging/initialize.rs

```diff
@@ -96,54 +94,72 @@ struct LoggingContext<'a, A: Allocate> {
-        ));
-
-        let errs = self
-            .worker
-            .dataflow_named("Dataflow: logging errors", |scope| {
+    fn construct_dataflows(&mut self) -> LoggingDataflows {
+        let dataflow_index = self.worker.next_dataflow_index();
+        self.worker.dataflow_named("Dataflow: logging", |scope| {
+            let mut collections = BTreeMap::new();
+
+            let super::timely::Return {
+                collections: timely_collections,
+                compute_events: compute_events_timely,
+            } = super::timely::construct(scope.clone(), self.config, self.t_event_queue.clone());
+            collections.extend(timely_collections);
+
+            collections.extend(super::reachability::construct(
+                scope.clone(),
+                self.config,
+                self.r_event_queue.clone(),
+            ));
+
+            let super::differential::Return {
+                collections: differential_collections,
+                compute_events: compute_events_differential,
+            } = super::differential::construct(
+                scope.clone(),
+                self.config,
+                self.d_event_queue.clone(),
+                Rc::clone(&self.shared_state),
+            );
+            collections.extend(differential_collections);
+
+            let super::compute::Return {
+                collections: compute_collections,
+            } = super::compute::construct(
+                scope.clone(),
+                scope.parent.clone(),
+                self.config,
+                self.c_event_queue.clone(),
+                [compute_events_timely, compute_events_differential],
+                Rc::clone(&self.shared_state),
+            );
+            collections.extend(compute_collections);
+
+            let errs = scope.scoped("Dataflow: logging errors", |scope| {
                 let collection: KeyCollection<_, DataflowError, Diff> =
                     Collection::empty(scope).into();
                 collection
                     .mz_arrange::<ErrBatcher<_, _>, ErrBuilder<_, _>, _>("Arrange logging err")
                     .trace
             });
 
-        // TODO(vmarcos): If we introduce introspection sources that would match
-        // type specialization for keys, we'd need to ensure that type specialized
-        // variants reach the map below (issue database-issues#6763).
-        collections
-            .into_iter()
-            .map(|(log, collection)| {
-                let bundle =
-                    TraceBundle::new(collection.trace, errs.clone()).with_drop(collection.token);
-                (log, (bundle, collection.dataflow_index))
-            })
-            .collect()
+            // TODO(vmarcos): If we introduce introspection sources that would match
+            // type specialization for keys, we'd need to ensure that type specialized
+            // variants reach the map below (issue database-issues#6763).
```

Comment:
Given that we've moved away from arrangement specialization, maybe this is a good time to remove this comment?